pkg/controllers/elasticsearch/configmap/resources.go

```diff
@@ -16,6 +34,49 @@ func esConfigConfigMap(c *v1alpha1.ElasticsearchCluster, np *v1alpha1.Elasticsea
 			OwnerReferences: []metav1.OwnerReference{util.NewControllerRef(c)},
 			Labels:          util.ClusterLabels(c),
 		},
-		Data: np.Config,
+		// TODO: move the environment variable names into a general purpose package
+		Data: map[string]string{
+			"elasticsearch.yml": generateConfig(c, np),
+		},
+	}
+}
+
+func generateConfig(c *v1alpha1.ElasticsearchCluster, np *v1alpha1.ElasticsearchClusterNodePool) string {
+	minimumMasters := c.Spec.MinimumMasters
+	if minimumMasters == 0 {
+		// auto-manage minimum master count
+		totalMasters := countMasterReplicas(c.Spec.NodePools)
+		minimumMasters = calculateQuorom(totalMasters)
+	}
+	return fmt.Sprintf(configTemplate, minimumMasters)
+}
+
+// TODO: move these functions into a shared package
```

Comment:
Yeah, I think we should do that before merging this.

Reply:
I've put something together for this, but it won't actually reduce duplication between validation and this package as they work on different types.

Validation is performed on internal types, and due to lack of generics, there's no way to share these functions (except for CalculateQuorum).

Reply:
I've pushed up my changes which include this. For now, I've kept the two functions that work on API types in validation, and also created `pkg/util/api` to house the v1alpha1 versions of these functions (which the `configmap` package now depends upon).

I did consider converting the types using the generated conversion functions, but this seems excessive for what is effectively a couple of for loops.